### ISSUE TITLE
Tying Word Vectors and Word Classifiers: A Loss Framework for Language Modeling

### ISSUE THREAD

**Issue by icoxfog417** [arXivTimes/arXivTimes#290]
## 一言でいうと

自然言語処理において単語の予測をクラス分類のように0/1でやるのは不自然だということで、予測分布の距離を加味することを提案。また、それにより入力のembeddingと出力のprojectionは使いまわしが可能になることを理論的に証明。これによりパラメーター量も下げられる、はず。

### 論文リンク

https://arxiv.org/abs/1611.01462

### 著者/所属機関

Hakan Inan, Khashayar Khosravi, Richard Socher

* Stanford University
* Salesforce Research

**Comment by icoxfog417** [comment]
Kerasで実装

https://github.com/icoxfog417/tying-wv-and-wc

**Comment by icoxfog417** [comment]
Input embeddin = Output embeddingにする効果は同時期のこちらの研究でも言及されていたらしい

* [Using the Output Embedding to Improve Language Models](https://arxiv.org/abs/1608.05859)

PyTorchの言語モデルではこのtyingの機能が実装されている

* [pytorch/examples/word_language_model/model](https://github.com/pytorch/examples/blob/1c6d9d276f3a0c484226996ab7f9df4f90ce52f4/word_language_model/model.py#L28)
